File: WebAdmin/classes/FeedbackRepository.js
$document.ready(function(){

    let clsFeedbackRepository = class {
       constructor (){
           const ReachUpWebAPI;
           ReachUpWebAPI = new clsReachUpWebAPI();
        }

        async getByDate(type, begin, end)
        {
           if (await ReachUpWebAPI.httpGet('Feedback/ByDate?type=' + type + '&begin=' + begin + '&end=' + end)) {
               return ReachUpWebAPI.response;
           }

           else {
               return false;
           }
        }

        async getByUser(email)
        {
           if (await ReachUpWebAPI.httpGet('Feedback/ByUser?email=' + email)) {
               return ReachUpWebAPI.response;
           }

           else {
               return false;
           }
        }

        async getByType(type)
        {
           if (await ReachUpWebAPI.httpGet('Feedback/GetAll?type=' + type)) {
               return ReachUpWebAPI.response;
           }

           else {
               return false;
           }
        }

        async get(id)
        {
           if (await ReachUpWebAPI.httpGet('Feedback/Get?id=' + id)) {
               return ReachUpWebAPI.response;
           }

           else {
               return false;
           }
        }

        async post(feedback)
        {
           if (await ReachUpWebAPI.httpPost('Feedback/Post')) {
               return true;
           }

           else {
               return false;
           }
        }

        async patch(feedback)
        {
           if (await ReachUpWebAPI.httpPatch('Feedback/Patch')) {
               return true;
           }

           else {
               return false;
           }
        }

        async delete(id)
        {
           if (await ReachUpWebAPI.httpDelete('Feedback/Delete?id=' + id)) {
               return true;
           }

           else {
               return false;
           }
        }

    } 

    const FeedbackRepository;
    FeedbackRepository = new clsFeedbackRepository("");

})

File: WebAdmin/Repositories/FeedbackRepository.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using WebAdmin.Models;
using System.Net.Http;

namespace WebAdmin.Repositories
{
    public class FeedbackRepository
    {
        private static readonly string BaseURL = clsApiConnection.BaseURL;
        private static readonly HttpClient client = new HttpClient();

        public async Task<List<Feedback>> ByDate(int type, string begin, string end)
        {
           string response = await client.GetAsync($"Feedback/ByDate?Type={type}&Begin={begin}&End={end}");
           List<Feedback> _feedbacks =
             JsonConvert.DeserializeObject<List<Feedback>>(response);
           return _feedbacks;
        }

        public async Task<List<Feedback>> ByUser(string email)
        {
           string response = await client.GetAsync($"Feedback/ByUser?Email={email}");
           List<Feedback> _feedbacks =
             JsonConvert.DeserializeObject<List<Feedback>>(response);
           return _feedbacks;
        }

        public async Task<List<Feedback>> GetAll(int type)
        {
           string response = await client.GetAsync($"Feedback/GetAll?Type={type}");
           List<Feedback> _feedbacks =
             JsonConvert.DeserializeObject<List<Feedback>>(response);
           return _feedbacks;
        }

        public async Task<Feedback> Get(int id)
        {
           string response = await client.GetAsync($"Feedback/Get?Id={id}");
           Feedback _feedback =
             JsonConvert.DeserializeObject<Feedback>(response);
           return _feedback;
        }

        public async Task<bool> Post(Feedback feedback)
        {
           bool response = await client.PostAsync($"Feedback/Post");
           return response;
        }

        public async Task<bool> Patch(Feedback feedback)
        {
           bool response = await client.PatchAsync($"Feedback/Patch");
           return response;
        }

        public async Task<bool> Delete(int id)
        {
           bool response = await client.DeleteAsync($"Feedback/Delete?Id={id}");
           return response;
        }
    }
}